### ISSUE TITLE
[Bug?] JEI not showing staged recipes after earning necessary stage

### ISSUE THREAD

**Issue by thephoenixlodge** [codetaylor/artisan-worktables#24]
As mentioned briefly in issue #22 I can't seem to get JEI to actually show the staged recipes I'm adding, though they are working and able to be crafted in the table.

Mod Versions:
Forge: 2590
Athaneum: 1.3.3
Artisan Worktables: 1.7.12
CraftTweaker: 4.0.12

Script used:
```
////Imports
import mods.artisanworktables.Worktable;

////Variables
val ironChestPlate = <minecraft:iron_chestplate>;
val ironChestPlateAdv = <minecraft:iron_chestplate>.withTag({AttributeModifiers: [{Slot:"chest",AttributeName:"generic.maxHealth",Name:"generic.maxHealth",Amount:5,Operation:0,UUIDLeast:123,UUIDMost:123}]});
val bsHammer = <ore:blacksmiths_hammer>;
val ingotIron = <ore:ingotIron>;
val plateIron = <ore:plateIron>;



//remove initially
recipes.remove(ironChestPlate);


//Add staged recipe in blacksmith table
Worktable.addStagedRecipeShaped("diagramIronChestBasic", "blacksmith", ironChestPlate, bsHammer, 1, false, [[ingotIron, null, ingotIron],[ingotIron, ingotIron, ingotIron],[ingotIron, ingotIron, ingotIron]]);

Worktable.addStagedRecipeShaped("diagramIronChestAdvanced", "blacksmith", ironChestPlateAdv, bsHammer, 1, false, [[ingotIron, null, ingotIron],[plateIron, ingotIron, plateIron],[ingotIron, ingotIron, ingotIron]]);
```

**Comment by codetaylor** [comment]
What version of GameStages are you using?

**Comment by codetaylor** [comment]
Also, what version of JEI?

**Comment by codetaylor** [comment]
Attempted to reproduce using these versions:
* Forge 2590
* Athaneum 1.3.3
* Artisan Worktables 1.7.12
* CraftTweaker 4.0.12.323
* JEI 4.8.5.146
* GameStages 1.0.75

Used this recipe script:
```
////Imports
import mods.artisanworktables.Worktable;

////Variables
val ironChestPlate = <minecraft:iron_chestplate>;
val ironChestPlateAdv = <minecraft:iron_chestplate>.withTag({AttributeModifiers: [{Slot:"chest",AttributeName:"generic.maxHealth",Name:"generic.maxHealth",Amount:5,Operation:0,UUIDLeast:123,UUIDMost:123}]});
val bsHammer = <ore:blacksmiths_hammer>;
val ingotIron = <ore:ingotIron>;
val plateIron = <ore:plankWood>; //<ore:plateIron>; // replaced with vanilla material for testing



//remove initially
recipes.remove(ironChestPlate);


//Add staged recipe in blacksmith table
Worktable.addStagedRecipeShaped("diagramIronChestBasic", "blacksmith", ironChestPlate, bsHammer, 1, false, [[ingotIron, null, ingotIron],[ingotIron, ingotIron, ingotIron],[ingotIron, ingotIron, ingotIron]]);

Worktable.addStagedRecipeShaped("diagramIronChestAdvanced", "blacksmith", ironChestPlateAdv, bsHammer, 1, false, [[ingotIron, null, ingotIron],[plateIron, ingotIron, plateIron],[ingotIron, ingotIron, ingotIron]]);
```

Steps:
1. Load MC and world
2. run command `/gamestage add @p diagramIronChestBasic`
3. Hover over Blacksmith's table and press `U`
4. Nothing from JEI is shown
5. Hover over Iron Chestplate and press `R`
6. No recipes in JEI are shown for the Blacksmith's table

**Comment by codetaylor** [comment]
Attempted to reproduce using the same versions as above, but used a different script:

```
import mods.artisanworktables.Worktable;

Worktable.addStagedRecipeShaped("one", "carpenter", <minecraft:crafting_table>, <ore:carpenters_hammer>, 3, false, [
    [<minecraft:planks>],
    [<minecraft:planks>],
    [<minecraft:planks>]]);
```

Steps:
1. Load MC and world
2. run command `/gamestage add @p one`
3. Hover over Carpenter's table and press `U`
4. Recipe is shown in JEI as expected

**Comment by codetaylor** [comment]
I used the same recipe as in https://github.com/codetaylor/artisan-worktables/issues/24#issuecomment-358050306 but I changed the stage name from `one` to `camelCase` and the recipe then failed to show in JEI when it was expected to be shown.

The point of failure lies with the case of gamestage names.

**Comment by codetaylor** [comment]
Interestingly enough, running the command:

```
/gamestage info
```

Displays:

```
one, diagramironchestbasic, camelcase
```

GameStages must `.toLowerCase()` stage names. I should be able to fix this by ensuring that recipe stage names are stored as their lowercase variant.

**Comment by codetaylor** [comment]
Well, would'ya [look at that](https://github.com/Darkhax-Minecraft/Game-Stages/blob/master/src/main/java/net/darkhax/gamestages/capabilities/PlayerDataHandler.java#L221-L230).

**Comment by codetaylor** [comment]
![image](https://user-images.githubusercontent.com/1240716/35005071-169537fa-faa7-11e7-8b7e-f9a1ea4b1678.png)

**Comment by codetaylor** [comment]
Fix available in 1.12.2-1.7.13

**Comment by codetaylor** [comment]
It turns out that this was also an issue in the new recipe system in my feature branch. Fixed!

Thank you very much for taking the time to make a bug report. :)
